appbuilder/services/npm-service.ts

```diff
@@ -75,6 +79,54 @@ export class NpmService implements INpmService {
 		}).future<void>()();
 	}
 
+	public search(projectDir: string, keywords: string[], args?: string[]): IFuture<IBasicPluginInformation[]> {
+		return ((): IBasicPluginInformation[] => {
+			let result: IBasicPluginInformation[] = [];
+			let spawnResult = this.executeNpmCommand(projectDir, args || [], keywords.join(" ")).wait();
```

Comment:
It appears that the argument `"search"` is missing here. If only keywords are passed you'd get an error here since you'd spawn `$ npm <keywords>` instead of `$ npm search <keywords>`